cudaungappedextender/samples/segalign_seedfilter.cu

```diff
@@ -0,0 +1,414 @@
+// integers
+__global__ void compress_rev_comp_string(char* dst_seq, char* src_seq,
+                                         uint32_t len){
+    ...}
+
+///////////////////// End Ungapped Extension related functions executed on the
+/// GPU ///////////////
+
+__global__ void find_num_hits(int num_seeds,
+                              const uint32_t* __restrict__ d_index_table,
+                              uint64_t* seed_offsets, uint32_t* seed_hit_num){
+    ...}
+
+__global__ void find_hits(const uint32_t* __restrict__ d_index_table,
+                          const uint32_t* __restrict__ d_pos_table,
+                          uint64_t* d_seed_offsets, uint32_t seed_size,
+                          uint32_t* seed_hit_num, int num_hits, Anchor* d_hit,
+                          uint32_t start_seed_index, uint32_t start_hit_index)
+{
+    ...
+}
+
+int InitializeProcessor(int num_gpu, bool transition, uint32_t WGA_CHUNK,
+                        uint32_t input_seed_size, int* sub_mat, int input_xdrop,
+                        int input_hspthresh, bool input_noentropy)
+{
+
+    int nDevices;
+
+    cudaError_t err = cudaGetDeviceCount(&nDevices);
+    if (err != cudaSuccess)
+    {
+        fprintf(stderr, "Error: No GPU device found!\n");
+        exit(1);
+    }
+
+    if (num_gpu == -1)
+    {
+        NUM_DEVICES = nDevices;
+    }
+    else
+    {
+        if (num_gpu <= nDevices)
+        {
+            NUM_DEVICES = num_gpu;
+        }
+        else
+        {
+            fprintf(stderr, "Requested GPUs greater than available GPUs\n");
+            exit(10);
+        }
+    }
+
+    fprintf(stderr, "Using %d GPU(s)\n", NUM_DEVICES);
+
+    seed_size = input_seed_size;
+
+    if (transition)
+        MAX_SEEDS = 13 * WGA_CHUNK;
+    else
+        MAX_SEEDS = WGA_CHUNK;
+
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, 0);
+    float global_mem_gb =
+        static_cast<float>(deviceProp.totalGlobalMem / 1073741824.0f);
+    MAX_SEED_HITS = global_mem_gb * MAX_SEED_HITS_PER_GB;
+
+    Anchor zeroHit;
+    zeroHit.query_position_in_read_  = 0;
+    zeroHit.target_position_in_read_ = 0;
+
+    ScoredSegment zeroHsp;
+    zeroHsp.anchor.query_position_in_read_  = 0;
+    zeroHsp.anchor.target_position_in_read_ = 0;
+    zeroHsp.len                             = 0;
+    zeroHsp.score                           = 0;
+
+    d_ref_seq      = (char**)malloc(NUM_DEVICES * sizeof(char*));
+    d_query_seq    = (char**)malloc(BUFFER_DEPTH * NUM_DEVICES * sizeof(char*));
+    d_query_rc_seq = (char**)malloc(BUFFER_DEPTH * NUM_DEVICES * sizeof(char*));
+
+    d_index_table = (uint32_t**)malloc(NUM_DEVICES * sizeof(uint32_t*));
+    d_pos_table   = (uint32_t**)malloc(NUM_DEVICES * sizeof(uint32_t*));
+
+    d_seed_offsets = (uint64_t**)malloc(NUM_DEVICES * sizeof(uint64_t*));
+
+    d_hit_num_array = (uint32_t**)malloc(NUM_DEVICES * sizeof(int32_t*));
```

Comment:
typo? `sizeof(uint32_t*)`, not that it really matters since they're the same size